addon/-private/system/model/internal-model.js

```diff
@@ -104,13 +104,12 @@ const {
   @class InternalModel
 */
 export default class InternalModel {
-  constructor(modelClass, id, store, data) {
+  constructor(modelName, id, store, data) {
     heimdall.increment(new_InternalModel);
```

Comment:
should the internalModel have a direct reference to its `classMeta` / `internalClass`, that way `get modelClass() ` can simply delegate to it.